lightning/src/util/mod.rs

```diff
@@ -49,3 +49,62 @@ pub mod test_utils;
 /// machine errors and used in fuzz targets and tests.
 #[cfg(any(test, feature = "fuzztarget", feature = "_test_utils"))]
 pub mod enforcing_trait_impls;
+
+pub(crate) mod crypto {
+	use bitcoin::hashes::{Hash, HashEngine};
+	use bitcoin::hashes::hmac::{Hmac, HmacEngine};
+	use bitcoin::hashes::sha256::Hash as Sha256;
+	use prelude::*;
+
+	pub fn hkdf_extract_expand_twice(salt: &[u8], ikm: &[u8]) -> ([u8; 32], [u8; 32]) {
+		let keys = hkdf_extract_expand(salt, ikm, 2);
+		(keys[0], keys[1])
+	}
+
+	pub fn hkdf_extract_expand_thrice(salt: &[u8], ikm: &[u8]) -> ([u8; 32], [u8; 32], [u8; 32]) {
+		let keys = hkdf_extract_expand(salt, ikm, 3);
+		(keys[0], keys[1], keys[2])
+	}
+
+	#[cfg(test)]
+	pub(super) fn hkdf_extract_expand_test(salt: &[u8], ikm: &[u8], num_keys: u8) -> Vec<[u8; 32]> {
+		hkdf_extract_expand(salt, ikm, num_keys)
+	}
+
+	fn hkdf_extract_expand(salt: &[u8], ikm: &[u8], num_keys: u8) -> Vec<[u8; 32]> {
```

Comment:
eek, can we avoid the vec here? We should be able to easily macro the funciton generation here, no?